nionswift_plugin/usim/ScanDevice.py

```diff
@@ -129,55 +276,30 @@ def read_partial(self, frame_number: int, pixels_to_skip: int) -> typing.Tuple[t
+                scan_data.append(self.get_scan_data(current_frame.frame_parameters, channel))
             current_frame.scan_data = scan_data
 
         is_synchronized_scan = frame_parameters.external_clock_mode != 0
-        if current_frame.data_count == 0 and is_synchronized_scan:
-            self.__instrument.live_probe_position = Geometry.FloatPoint()
 
         target_count = 0
-        while self.__is_scanning and target_count <= current_frame.data_count:
-            if is_synchronized_scan:
-                # set the probe position
-                h, w = current_frame.scan_data[0].shape
-                # calculate relative position within sub-area
-                ry, rx = current_frame.data_count // w / h - 0.5, current_frame.data_count % w / w - 0.5
-                # now translate to context
-                ss = Geometry.FloatSize.make(frame_parameters.subscan_fractional_size) if frame_parameters.subscan_fractional_size else Geometry.FloatSize(h=1.0, w=1.0)
-                oo = Geometry.FloatPoint.make(frame_parameters.subscan_fractional_center) - Geometry.FloatPoint(y=0.5, x=0.5) if frame_parameters.subscan_fractional_center else Geometry.FloatPoint()
-                oo += Geometry.FloatSize(h=frame_parameters.center_nm[0] / frame_parameters.fov_nm, w=frame_parameters.center_nm[1] / frame_parameters.fov_nm)
-                pt = Geometry.FloatPoint(y=ry * ss.height + oo.y, x=rx * ss.width + oo.x)
-                pt = pt.rotate(frame_parameters.rotation_rad)
-                if frame_parameters.subscan_rotation:
-                    pt = pt.rotate(-frame_parameters.subscan_rotation, oo)
-                # print(f"{x}, {y} = ({ry} - 0.5) * {ss.height} + {oo.y} / (ry - 0.5) * ss.height + oo.y")
-                # >>> def f(s, c, x): return (x - 0.5) * s + c # |------<---c--->------------|
-                self.__instrument.live_probe_position = pt + Geometry.FloatPoint(y=0.5, x=0.5)
-                # do a synchronized readout
-                if current_frame.data_count % size.width == 0:
-                    # throw away two flyback images at beginning of line
-                    if not self.__is_scanning or not self.__instrument.wait_for_camera_frame(frame_parameters.external_clock_wait_time_ms / 1000):
-                        current_frame.bad = True
-                        current_frame.complete = True
-                    if not self.__is_scanning or not self.__instrument.wait_for_camera_frame(frame_parameters.external_clock_wait_time_ms / 1000):
-                        current_frame.bad = True
-                        current_frame.complete = True
-                if not self.__is_scanning or not self.__instrument.wait_for_camera_frame(frame_parameters.external_clock_wait_time_ms / 1000):
-                    current_frame.bad = True
-                    current_frame.complete = True
-                target_count = current_frame.data_count + 1
-            else:
+        if is_synchronized_scan:
```

Comment:
Comments would be helpful here, even though they weren't in the old code.

Generally, in these long functions with lots of `if` statements, more comments are helpful, even if they're obvious.